Indexing/TermSubstitutionTree.hpp

```diff
@@ -54,77 +56,54 @@ class TermSubstitutionTree
    */
   TermSubstitutionTree(bool useC=false, bool replaceFunctionalSubterms = false, bool extra = false);
 
-  void insert(TermList t, Literal* lit, Clause* cls);
-  void remove(TermList t, Literal* lit, Clause* cls);
-  void insert(TermList t, TermList trm);
-  void insert(TermList t, TermList trm, Literal* lit, Clause* cls);
+  void handle(TypedTermList tt, Literal* lit, Clause* cls, bool insert);
```

Comment:
A lot of these functions are now trivial and the implementation could be inlined here.